code/game/objects/random/food.dm

```diff
@@ -45,8 +44,10 @@
 	icon_state = "food-red"
 
 /obj/random/pizza/item_to_spawn()
-	for(var/pizza in subtypesof(/obj/item/weapon/reagent_containers/food/snacks/sliceable/pizza))
-		return pizza
+	return pickweight(list(/obj/item/weapon/reagent_containers/food/snacks/sliceable/pizza/margherita = 1,
```

Comment:
subtypesof(/obj/item/weapon/reagent_containers/food/snacks/sliceable/pizza) returns a list() which could be put in pickweight() as a valid function